src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Logging/LspLogMessageLogger.cs

```diff
@@ -55,30 +49,39 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
                 message += " " + exceptionString;
         }
 
-        if (message != null && logLevel != LogLevel.None)
+        string messagePrefix = "";
+
+        _externalScopeProvider?.ForEachScope((scope, _) =>
         {
-            message = $"[{categoryName}] {message}";
-            try
+            if (scope is LspLoggingScope lspLoggingScope)
```

Comment:
Although nothing prevents us from just adding a string as the scope, I decided to a bit more strict about the typing here.